appserver/payara-appserver-modules/microprofile/fault-tolerance/src/main/java/fish/payara/microprofile/faulttolerance/validators/FallbackValidator.java

```diff
@@ -65,32 +68,34 @@
     public static void validateAnnotation(Fallback fallback, AnnotatedMethod<?> annotatedMethod, Config config) 
             throws ClassNotFoundException, NoSuchMethodException {
         // Get the fallbackMethod
-        String fallbackMethod = (String) FaultToleranceCdiUtils.getOverrideValue(
+        String fallbackMethod = FaultToleranceCdiUtils.getOverrideValue( 
                 config, Fallback.class, "fallbackMethod", annotatedMethod.getJavaMember().getName(), 
                 annotatedMethod.getJavaMember().getDeclaringClass().getCanonicalName(), String.class)
                 .orElse(fallback.fallbackMethod());
         
         // Get the fallbackClass, and check that it can be found
-        Class<? extends FallbackHandler> fallbackClass = (Class<? extends FallbackHandler>) Thread.currentThread()
-                .getContextClassLoader().loadClass((String) FaultToleranceCdiUtils
+        Optional<String> fallbackClassName = FaultToleranceCdiUtils
                 .getOverrideValue(config, Fallback.class, "value", annotatedMethod.getJavaMember().getName(), 
-                annotatedMethod.getJavaMember().getDeclaringClass().getCanonicalName(), String.class)
-                .orElse(fallback.value().getName()));
-        
+                annotatedMethod.getJavaMember().getDeclaringClass().getCanonicalName(), String.class);
+        @SuppressWarnings("unchecked")
+        Class<? extends FallbackHandler<?>> fallbackClass = fallbackClassName.isPresent() 
+            ? (Class<? extends FallbackHandler<?>>) Thread.currentThread().getContextClassLoader()
+                    .loadClass(fallbackClassName.get())
+            : fallback.value(); 
```

Comment:
Note same fix for `Class` loading as above with `String` class name from config.